src/xref2/compile.ml

```diff
@@ -353,30 +371,35 @@ and functor_parameter : Env.t -> FunctorParameter.t -> FunctorParameter.t =
 and functor_parameter_parameter :
     Env.t -> FunctorParameter.parameter -> FunctorParameter.parameter =
  fun env' a ->
-  let exception Compile_functor_parameter_parameter in
-
-  let functor_arg, env =
-    match Env.lookup_module' a.id env' with
-    | Some (_, _ as x) -> x
-    | None -> raise Compile_functor_parameter_parameter
+  let open Utils.ResultMonad in
+  let get_module_type_expr = function
+    | Component.Module.ModuleType expr -> Ok expr
+    | _ -> Error `Resolve_type
   in
-  let env, expn =
+  match
+    Env.lookup_module' a.id env' |> of_option ~error:`Lookup
```

Comment:
There should basically never be a failure from `Env.lookup_module` at this point, it indicates an internal error.

Reply:
The types can't ensures that, should we throw a fatal error instead ?

Reply:
I think if get it in as it is, we can decide later on which errors _should_ be fatal